src/nspawn/nspawn.c

```diff
@@ -2517,19 +2517,15 @@ static int setup_propagate(const char *root) {
         p = strjoina("/run/systemd/nspawn/propagate/", arg_machine);
         (void) mkdir_p(p, 0600);
 
-        r = userns_mkdir(root, "/run/systemd", 0755, 0, 0);
+        r = userns_mkdir(root, "/run/host", 0755, 0, 0);
         if (r < 0)
-                return log_error_errno(r, "Failed to create /run/systemd: %m");
+                return log_error_errno(r, "Failed to create /run/host: %m");
 
-        r = userns_mkdir(root, "/run/systemd/nspawn", 0755, 0, 0);
+        r = userns_mkdir(root, "/run/host/incoming", 0600, 0, 0);
         if (r < 0)
-                return log_error_errno(r, "Failed to create /run/systemd/nspawn: %m");
+                return log_error_errno(r, "Failed to create /run/host/incoming: %m");
 
-        r = userns_mkdir(root, "/run/systemd/nspawn/incoming", 0600, 0, 0);
-        if (r < 0)
-                return log_error_errno(r, "Failed to create /run/systemd/nspawn/incoming: %m");
-
-        q = prefix_roota(root, "/run/systemd/nspawn/incoming");
+        q = prefix_roota(root, "/run/host/incoming");
```

Comment:
I wonder if selinux will barf on this. But let's wait and see.

Reply:
it will, at the bare minimum it will need something like:

```
--- a/policy/modules/system/systemd.fc
+++ b/policy/modules/system/systemd.fc
@@ -72,6 +72,7 @@
 /run/systemd/users(/.*)?       gen_context(system_u:object_r:systemd_logind_runtime_t,s0)
 /run/systemd/inhibit(/.*)?     gen_context(system_u:object_r:systemd_logind_inhibit_runtime_t,s0)
 /run/systemd/nspawn(/.*)?      gen_context(system_u:object_r:systemd_nspawn_runtime_t,s0)
+/run/host/incoming(/.*)?       gen_context(system_u:object_r:systemd_nspawn_runtime_t,s0)
 /run/systemd/machines(/.*)?    gen_context(system_u:object_r:systemd_machined_runtime_t,s0)
 /run/systemd/netif(/.*)?       gen_context(system_u:object_r:systemd_networkd_runtime_t,s0)
 

```